server/routes/monitors.js

```diff
@@ -13,54 +13,115 @@
  *   permissions and limitations under the License.
  */
 
-export default function (server, services) {
+import { schema } from '@kbn/config-schema';
+
+export default function (services, router) {
   const { monitorService } = services;
 
-  server.route({
-    path: '/api/alerting/monitors',
-    method: 'GET',
-    handler: monitorService.getMonitors,
-  });
+  router.get(
+    {
+      path: '/api/alerting/monitors',
+      validate: {
+        query: schema.object({
+          from: schema.number(),
+          size: schema.number(),
+          search: schema.string(),
+          sortField: schema.string(),
+          sortDirection: schema.string(),
+          state: schema.string(),
```

Comment:
Are these fields always provided in Kibana?

Reply:
There is only one place calling this API, which aims to get all the monitors to show in the dashboard. https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/blob/32e202fb4add9e7e78f4a9855a6510321e6e177b/public/pages/Monitors/containers/Monitors/Monitors.js#L142 . The fields are all provided.
Btw, I write down all the field just make the validation to be accurate, we could use `{ query: schema.any()}` at anytime to pass the validation.